textworld/challenges/spaceship/textworld_data/logic/CPU.twl

```diff
@@ -0,0 +1,60 @@
+# CPU-Like
+type cpu : o {
+    predicates {
+        #turned_on(cpu);
+        #turned_off(cpu);
+
+        read/e(cpu);
+        unread/e(cpu);
+    }
+
+    rules {
+        check/e     :: $at(P, r) & $at(s, r) & $on(cpu, s) & unread/e(cpu) -> read/e(cpu);
+        examine/cpu :: at(P, r) & $at(s, r) & $on(cpu, s) -> at(P, r);  # Nothing changes.
+    }
+
+    reverse_rules {
+        check/e     :: check/e;
```

Comment:
Well my understanding about this was a bit different! Let me describe it, & pls correct me if I'm wrong:
I assumed that the reverse actions provides the return to previous state as ell as repetition of an action (e.g. examine/door :: examine/door). Since we need to have check email in the "available actions" array whenever the player is in Sleeping Station, I decided to have return rule for check/e. What do you think?